src/Agent/NewRelic/Profiler/Profiler/ICorProfilerCallbackBase.h

```diff
@@ -797,22 +797,34 @@ namespace Profiler {
 // Ideally we need to be able to handle file paths with multibyte characters in them. Currently, we do not.
 #pragma warning(push)
 #pragma warning(disable : 4244)
-            // attempt to open the file
-            xifstream fileStream(std::string(filePath.begin(), filePath.end()), std::ios::in);
+
+            // Open file to read as a binary byte stream.  This lets us reliably detect and
+            // remove a UTF-8 BOM at the beginning of the file.
+            // See https://github.com/newrelic/newrelic-dotnet-agent/issues/267 for context
+            ifstream inFile(std::string(filePath.begin(), filePath.end()), std::ios::binary);
```

Comment:
Is there a reason why we are no longer using the `x` macro for some of the type here .ie `ifstream`, `ostringstream` ... . I'm not expert in this but, my impression was the `x` macro ensure the cross platform behavior according [this](https://github.com/newrelic/newrelic-dotnet-agent/blob/5b375b3c38517c8326aac0a70a4901f99f1a483c/src/Agent/NewRelic/Profiler/Common/xplat.h#L17)

Reply:
Discussed with @nr-ahemsath on this. The reason why this code didn't use `x` macro was UTF-8 BOM can only be detected in a binary byte (char) stream in Linux.